content/blog/manage-infrastructure-with-pac/index.md

```diff
@@ -0,0 +1,233 @@
+4. [Apply a Policy Pack]({{ < relref “docs/get-started/crossguard/enforcing-a-policy-pack” >}}) across an organization to validate all the infrastructure deployed
+
+CrossGuard ensures that you can enforce best practices for cost, compliance, security, and team practices for a single project or across your organization. Let’s look at how we can apply policies to infrastructure deployed across cloud providers and Kubernetes.
+
+## Controlling cost on AWS
+
+The amount of an AWS monthly bill has become a cliché in the tech industry. From services using improperly sized AMIs to orphaned infrastructure left running, there are many ways to incur a huge bill. One way to limit infrastructure spend is to set a `not to exceed` policy. We want to calculate the costs before deployment. The pricing data is available online, but calculating the cost using configuration data to enumerate resources and services can be a manual process. However, if your infrastructure as code solution is Pulumi, you can use your preferred programming language to calculate the cost before deployment and check it against your policy.
+
+The example below shows how the policy finds all the instances and validates the aggregate cost against the maximum monthly amount allowed. It does this by finding the monthly on-demand price and aggregating all the costs.
+
+```ts
+import * as aws from "@pulumi/aws";
+import { PolicyPack, validateStackResourcesOfType } from "@pulumi/policy";
+import { getMonthlyOnDemandPrice, formatAmount } from "./utils";
+import { maxMonthlyCost } from "./config";
+
+new PolicyPack("aws", {
+    policies: [
+        {
+            name: "instance-cost-estimate",
+            description: `Limit instance costs to $${maxMonthlyCost}.`,
+            enforcementLevel: "mandatory",
+            validateStack: validateStackResourcesOfType(aws.ec2.Instance, (instances, args, reportViolation) => {
+                // Aggregate costs.
+                let totalMonthlyAmount = 0;
+                instances.forEach(instance => {
+                    totalMonthlyAmount += getMonthlyOnDemandPrice(instance.instanceType);
+                });
+
+                if (totalMonthlyAmount > maxMonthlyCost) {
+                    reportViolation(`Estimated monthly cost [${formatAmount(totalMonthlyAmount)}] exceeds [${formatAmount(maxMonthlyCost)}].`);
+                }
+            }),
+        }
+    ],
+});
+```
+
+Setting the maximum amount and calculating the monthly on-demand price for all `ec2` instances is aided by two helper classes `config` and `utils`. You can set variables such as `maxMonthlyCost` in `config.ts`. The utils can calculate costs from either a static file of prices, as demonstrated in the example, or through the Amazon Pricing API. Because we’re using a modern programming language, we can go beyond what a policy encoded in YAML or JSON can do.
```

Comment:
> The utils can calculate costs from either a static file of prices, as demonstrated in the example

Is it possible to provide a link to where to get this data?

Reply:
added link: https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/price-changes.html